Add a full-route key to the request map

liamchzh opened the pull request:
This provides access to the full route that contains the common prefix.
The full-route info is added to the ':compojure/full-route' key in the
request map.

liamchzh commented:
~~Couldn't add backplane as the reviewer, so pint you folks here in the comment @circleci/backplane~~

I'd also provide more context and rationale when opening the PR to the real Compojre project. Input is welcome!

liamchzh committed:
Add a full-route key to the request map

This provides access to the full route that contains the common prefix.
The full-route info is added to the ':compojure/full-route' key in the
request map.

Co-authored-by: Liam Chen <liamchzh@gmail.com>
Co-authored-by: Claire Alvis <claire.alvis@gmail.com>

liamchzh commented:
> I'd also provide more context and rationale when opening the PR to the real Compojre project.

We use `:compojure/route` to get the route info and add it to the attributes of our metrics and traces, but when `compojure.core/context` is used, we are not able to get the parameters that are not instantiated. This PR adds `:compojure/route` key that serves similarly as the existing `:compojure/route` does - the only difference is that the new key has a common prefix.

calvis commented on src/compojure/core.clj:
```diff
@@ -257,27 +265,39 @@
+           (clout/route-compile (str route# ":__path-info") ~re-context)
+           (if (string? route#)
+             {:context-route route#}
+             {}))))))
```
in the PR, you should mention that the `with-meta` is our attempt to not change the type of `make-context`, as it is public (albeit not documented). we could make a more direct map-based implementation if backwards compatibility of `make-context` is not important

also not entirely sure what the full syntax is of `context`, so it's possible we're missing a case

emilywoods reviewed (approved):
After talking through these changes, the pr looks good to me

liamchzh committed:
pass the path through to the context-request function

This approach is simpler, pass the path down, rather than
attaching it to the meta.

liamchzh commented:
https://github.com/weavejester/compojure/pull/212 has been merged!
